### ISSUE TITLE
Add unit test for shardingsphere-sharding

### ISSUE THREAD

**Issue by soulasuna** [apache/shardingsphere#10968]
Hi community,

This issue is for #10887.

## Aim
Add unit test for shardingsphere-sharding project to test its public functions.

## Basic Qualifications
- Java  
- Maven  
- Junit.Test

## Classes

Add unit test for classes to test its public functions.

- [ ] ShardingTableMetaDataBuilder
- [ ] ShardingSQLRewriteContextDecorator
- [ ] ShardingGeneratedKeyInsertValueParameterRewriter
- [ ] ShardingPaginationParameterRewriter
- [ ] ShardingParameterRewriterBuilder
- [ ] GeneratedKeyAssignmentTokenGenerator
- [ ] GeneratedKeyForUseDefaultInsertColumnsTokenGenerator
- [ ] GeneratedKeyInsertColumnTokenGenerator
- [ ] GeneratedKeyInsertValuesTokenGenerator
- [ ] AggregationDistinctTokenGenerator
- [ ] ConstraintTokenGenerator
- [ ] DistinctProjectionPrefixTokenGenerator
- [ ] IndexTokenGenerator
- [ ] OffsetTokenGenerator
- [ ] OrderByTokenGenerator
- [ ] ProjectionsTokenGenerator
- [ ] RowCountTokenGenerator
- [ ] ShardingInsertValuesTokenGenerator
- [ ] TableTokenGenerator

## Examples FYI
- AlgorithmProvidedShardingRuleBuilderTest

**Comment by fwhdzh** [comment]
Hello, I want to try this issue.

**Comment by fwhdzh** [comment]
Hello. For the last three class. I find that ShardingTableMetaDataBuilder has haved its unit test.  And I think ShardingSQLRewriteContextDecorator and ShardingParameterRewriterBuilder may have some problem since it looks like ShardingParameterRewriterBuilder.setUpParameterRewriters() has some "if" statement that will never be executed and ShardingSQLRewriteContextDecorator use ShardingParameterRewriterBuilder in its function.

**Comment by soulasuna** [comment]
@fwhdzh Hi, thank you for your contribution. yes can you fix them.

**Comment by tristaZero** [comment]
Hi, so it is still in progress, right?

**Comment by fwhdzh** [comment]
> Hi, so it is still in progress, right?

Yes, it is still in progress. I am sorry for that I am too busy these days to do this work. If it is urgent, please give this task to another person. I could look it the next week if possible.

**Comment by menghaoranss** [comment]
I will move the unfinished tasks to a new issue and close this issue now
